### ISSUE TITLE
Suggest Mapping / create Mapping

### ISSUE THREAD

**Issue by ryantheleach** [DemonWav/MinecraftDevIntelliJ#223]
This isn't a fully formed idea, but what about something that can create mappings / suggest mappings / generate mapping patch files for submission to the various mapping projects?

**Comment by kashike** [comment]
In the case of MCP, this would be either:
- a list of commands to be ran with MCPBot, assuming none of the fields/methods/parameters are not already named
- giving you something to paste into an [issue](https://github.com/ModCoderPack/MCPBot-Issues/issues) for already-named fields/methods/parameters

In addition to that, some things cannot be renamed (enum constant names, locked fields/methods/parameters).

**Comment by DenWav** [comment]
I wasn't sure what this meant initially, but yes some sort of integration with MCPBot is a "nice to have" feature that I haven't figured out how to properly handle yet. Unfortunately bspkrs isn't working on MCPBot anymore, so chances of a REST API to easily facilitate this are low.

Pre-formatting a command to run would be a less-nice but still probably helpful option. Same goes for automatic issue creation.

**Comment by ryantheleach** [comment]
I still think this would be cool, even if low priority. Is this enough of a bump for stalebot?

**Comment by DenWav** [comment]
This really is dependent on ModCoderPack/MCPBot-Issues#50. I can kind of recreate the behavior sort of just through reading the srg maps, but I wouldn't have access to any of the history of the names, nor the comment information or who changed it. Also wouldn't have access to specific version information. Connecting to IRC as an IRC bot inside IntelliJ is just ridiculous, so that's out of the question. Unfortunately the only API to interacting with MCPBot is through the IRC bot. It's open source, so theoretically someone could write a RESTful frontend, but the data isn't publicly available for hosting either, so you'd still need to convince the MCPBot folks to host this API.

Basically what I'm saying is this probably isn't going to happen.

**Comment by bspkrs** [comment]
I'm still active, just lazy :p

**Comment by DenWav** [comment]
I've created a new project, [mcpapi](https://github.com/DemonWav/mcpapi) that would provide this functionality. It's currently WIP, but once it's working and deployed it will allow this feature request to actually happen.
